FlyingSocks/Sources/AsyncSocket.swift

```diff
@@ -237,6 +268,22 @@ public struct AsyncSocketSequence: AsyncSequence, AsyncIteratorProtocol, Sendabl
     }
 }
 
+public struct AsyncSocketMessageSequence: AsyncSequence, AsyncIteratorProtocol, Sendable {
+    public typealias Element = (sockaddr_storage, [UInt8])
+
+    let socket: AsyncSocket
+
```

Comment:
Maybe a public init would be helpful so clients iterate with `for try await` but also adjust the default size (I assume smaller max size would be faster)

```swift
public init(socket: AsyncSocket, maxMessageLength: Int) { .. }

public mutating func next() async throws -> Element? {
    return try await socket.receive(atMost: maxMessageLength)
}
```